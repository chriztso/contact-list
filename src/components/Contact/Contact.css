.contact{
    display: flex;
    border: 1px solid rgb(233, 234, 240);
    align-items: center;
    height: 44px;
    position: relative;
}

.contact:hover{
    background-color: rgb(250,251,255);
}

.contact__checkbox{
    margin-left: 12px;
    margin-right: 12px;
}

.contact__name{
    display: flex;
    width: 20%; 
    align-items: center;
    color: rgb(68, 107, 222);
    font-weight: 600;
}

.contact__initials{
    height: 25px;
    width: 25px;
    background-color: rgb(232,224,252);
    color: black;
    font-weight: bold;
    border-radius: 50%;
    display: flex;
    align-items: center;
    margin-right: 5px;
    justify-content: center;
    font-size: 12px;
} 

.contact__value{
    width: 15%; 
}

.contact__location{
    width: 15%; 
}

.contact__deals{
    width: 10%; 
}

.contact__tags{
    width: 26%; 
}

.contact__email-button{
    position: absolute;
    right: 12px;
    display: flex;
    border: 1px solid rgb(207, 211, 223);
    border-radius: 5px;
    align-items: center;
    height: 30px;
}

.contact__email{
    border-right: 1px solid rgb(207, 211, 223);
    display: flex; 
    align-items: center;
    height: 100%;
    padding: 0px 10px;
    font-weight: 500;
}

.contact__down-arrow{
    display: flex; 
}