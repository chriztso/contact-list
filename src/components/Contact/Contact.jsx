import React, { useState } from 'react'
import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown';

import './Contact.css'

const Contact = (props) => {
    const {
        name,
        initials,
        contactTagsStr,
        numberOfDeals,
        dealsTotalValue,
        geoAddressStr
    } = props

    const [showButton, setShowButton] = useState(false)

    return (
        <div className="contact" 
          onMouseEnter={() => {setShowButton(true)}}
          onMouseLeave={() => {setShowButton(false)}}
        >
            <input type="checkbox" className="contact__checkbox"></input>
            <div className="contact__name">
                <div className="contact__initials">
                    {initials}
                </div>
                <div>{name}</div>
            </div>
            <span className="contact__value">${dealsTotalValue}</span>
            <span className="contact__location">{geoAddressStr}</span>
            <span className="contact__deals">{numberOfDeals}</span>
            <span className="contact__tags">{contactTagsStr}</span>
            {showButton && <div className="contact__email-button">
              <div className="contact__email">Email</div>
              <div className="contact__down-arrow"><ArrowDropDownIcon/></div>
            </div>}
        </div>
    )
}

export default Contact