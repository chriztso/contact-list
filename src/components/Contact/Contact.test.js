import { shallow } from 'enzyme';
import Contact from '../Contact/Contact'

describe('Contact', () => {
    const name = "John Smith"
    const initials = "JS"
    const contactTagsStr = "test"
    const numberOfDeals = "2"
    const dealsTotalValue = "100"
    const geoAddressStr = 'N/A'

    it('renders Contact', () => {
        const wrapper = shallow(
          <Contact 
            name={name}
            initials={initials}
            contactTagsStr={contactTagsStr}
            numberOfDeals={numberOfDeals}
            dealsTotalValue={dealsTotalValue}
            geoAddressStr={geoAddressStr}
          />);
        expect(wrapper).toMatchSnapshot();
    })

    it('renders checkbox', () => {
        const wrapper = shallow(
          <Contact 
            name={name}
            initials={initials}
            contactTagsStr={contactTagsStr}
            numberOfDeals={numberOfDeals}
            dealsTotalValue={dealsTotalValue}
            geoAddressStr={geoAddressStr}
          />);
        expect(wrapper.find('.contact__checkbox')).toHaveLength(1)
    } )

    it('renders initials', () => {
      const wrapper = shallow(
        <Contact 
          name={name}
          initials={initials}
          contactTagsStr={contactTagsStr}
          numberOfDeals={numberOfDeals}
          dealsTotalValue={dealsTotalValue}
          geoAddressStr={geoAddressStr}
        />);
      expect(wrapper.find('.contact__initials').text()).toEqual("JS")
    })

    it('renders value of deals', () => {
      const wrapper = shallow(
        <Contact 
          name={name}
          initials={initials}
          contactTagsStr={contactTagsStr}
          numberOfDeals={numberOfDeals}
          dealsTotalValue={dealsTotalValue}
          geoAddressStr={geoAddressStr}
        />);
      expect(wrapper.find('.contact__value').text()).toEqual("$100")
    })

    it('renders location', () => {
      const wrapper = shallow(
        <Contact 
          name={name}
          initials={initials}
          contactTagsStr={contactTagsStr}
          numberOfDeals={numberOfDeals}
          dealsTotalValue={dealsTotalValue}
          geoAddressStr={geoAddressStr}
        />);
      expect(wrapper.find('.contact__location').text()).toEqual("N/A")
    })

    it('renders number of deals', () => {
      const wrapper = shallow(
        <Contact 
          name={name}
          initials={initials}
          contactTagsStr={contactTagsStr}
          numberOfDeals={numberOfDeals}
          dealsTotalValue={dealsTotalValue}
          geoAddressStr={geoAddressStr}
        />);
      expect(wrapper.find('.contact__deals').text()).toEqual("2")
    })

    it('renders contact tags', () => {
      const wrapper = shallow(
        <Contact 
          name={name}
          initials={initials}
          contactTagsStr={contactTagsStr}
          numberOfDeals={numberOfDeals}
          dealsTotalValue={dealsTotalValue}
          geoAddressStr={geoAddressStr}
        />);
      expect(wrapper.find('.contact__tags').text()).toEqual("test")
    })
})