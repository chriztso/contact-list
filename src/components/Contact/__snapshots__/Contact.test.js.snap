// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact renders Contact 1`] = `
<div
  className="contact"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <input
    className="contact__checkbox"
    type="checkbox"
  />
  <div
    className="contact__name"
  >
    <div
      className="contact__initials"
    >
      JS
    </div>
    <div>
      John Smith
    </div>
  </div>
  <span
    className="contact__value"
  >
    $
    100
  </span>
  <span
    className="contact__location"
  >
    N/A
  </span>
  <span
    className="contact__deals"
  >
    2
  </span>
  <span
    className="contact__tags"
  >
    test
  </span>
</div>
`;
